### PR TITLE
use of checkers on docker_cli_restart_test.go

### PR DESCRIPTION
**Pull request by echo33**
Signed-off-by: weiyan weiyan3@huawei.com

### CODE REVIEW

File: integration-cli/docker_cli_restart_test.go [hunk 1]
```diff
@@ -15,16 +15,12 @@ func (s *DockerSuite) TestRestartStoppedContainer(c *check.C) {
 	dockerCmd(c, "wait", cleanedContainerID)
 
 	out, _ = dockerCmd(c, "logs", cleanedContainerID)
-	if out != "foobar\n" {
-		c.Errorf("container should've printed 'foobar'")
-	}
+	c.Assert(out, check.Equals, "foobar\n", check.Commentf("container should've printed 'foobar'"))
```
Comment (HuKeping) [inline integration-cli/docker_cli_restart_test.go]: Would you mind if using `checker.Equals` here

File: integration-cli/docker_cli_restart_test.go [hunk 2]
```diff
@@ -36,9 +32,7 @@ func (s *DockerSuite) TestRestartRunningContainer(c *check.C) {
 	c.Assert(waitRun(cleanedContainerID), check.IsNil)
 
 	out, _ = dockerCmd(c, "logs", cleanedContainerID)
-	if out != "foobar\n" {
-		c.Errorf("container should've printed 'foobar'")
-	}
+	c.Assert(out, check.Equals, "foobar\n", check.Commentf("container should've printed 'foobar'"))
```
Comment (cpuguy83) [inline integration-cli/docker_cli_restart_test.go]: Just `c.Assert(out, check.Equals, "foobar\n")` should suffice. The error message on failure will give the needed information in this cases.

File: integration-cli/docker_cli_restart_test.go [hunk 3]
```diff
@@ -46,9 +40,7 @@ func (s *DockerSuite) TestRestartRunningContainer(c *check.C) {
 
 	c.Assert(waitRun(cleanedContainerID), check.IsNil)
 
-	if out != "foobar\nfoobar\n" {
-		c.Errorf("container should've printed 'foobar' twice")
-	}
+	c.Assert(out, check.Equals, "foobar\nfoobar\n", check.Commentf("container should've printed 'foobar' twice"))
```
Comment (cpuguy83) [inline integration-cli/docker_cli_restart_test.go]: Same, no need for the Commentf here.

File: integration-cli/docker_cli_restart_test.go [hunk 4]
```diff
@@ -58,10 +50,8 @@ func (s *DockerSuite) TestRestartWithVolumes(c *check.C) {
 
 	cleanedContainerID := strings.TrimSpace(out)
 	out, _ = dockerCmd(c, "inspect", "--format", "{{ len .Mounts }}", cleanedContainerID)
-
-	if out = strings.Trim(out, " \n\r"); out != "1" {
-		c.Errorf("expect 1 volume received %s", out)
-	}
+	out = strings.Trim(out, " \n\r")
+	c.Assert(out, check.Equals, "1", check.Commentf("expect 1 volume received %s", out))
```
Comment (cpuguy83) [inline integration-cli/docker_cli_restart_test.go]: Same.

**Comment by cpuguy83** [comment]
Anywhere we are passing in `out` to a checker, there should be no reason to include it in the Commentf since this will be duplicated information in the error output.

**Comment by echo33** [comment]
@cpuguy83 thanks

**Comment by cpuguy83** [comment]
LGTM

For the cases of "\r\n", this can be "strings.TrimSpace".
I understand what's there is already there.

**Comment by vdemeester** [comment]
LGTM 🐶

**Comment by echo33** [comment]
@vdemeester thanks

**Pull request merged**